langchain/src/document_loaders/web/notionapi.ts

```diff
@@ -137,74 +138,167 @@ export class NotionAPILoader extends BaseDocumentLoader {
+      case "number": {
+        const propNumber = prop[prop.type];
+        return propNumber !== null ? propNumber.toString() : "";
+      }
+      case "url":
+        return prop[prop.type] || "";
+      case "select":
+        return prop[prop.type]?.name ?? "";
+      case "multi_select":
+        return `[${prop[prop.type].map((v) => `"${v.name}"`).join(", ")}]`;
+      case "status":
+        return prop[prop.type]?.name ?? "";
+      case "date":
+        return `${prop[prop.type]?.start ?? ""}${
+          prop[prop.type]?.end ? ` - ${prop[prop.type]?.end}` : ""
+        }`;
+      case "email":
+        return prop[prop.type] || "";
+      case "phone_number":
+        return prop[prop.type] || "";
+      case "checkbox":
+        return prop[prop.type].toString();
+      case "files":
+        return `[${prop[prop.type].map((v) => `"${v.name}"`).join(", ")}]`;
+      case "created_by":
+        return `["${prop[prop.type].object}", "${prop[prop.type].id}"]`;
+      case "created_time":
+        return prop[prop.type];
+      case "last_edited_by":
+        return `["${prop[prop.type].object}", "${prop[prop.type].id}"]`;
+      case "last_edited_time":
+        return prop[prop.type];
+      // TODO: BLOCKER: Not all formula types are supported in the SDK (dependant on Notion)
+      // case "formula": {
+      //   const formulaProp = prop[prop.type];
+      //   switch (formulaProp.type) {
+      //     case "boolean":
+      //       const formulaBool = formulaProp[formulaProp.type];
+      //       return formulaBool !== null ? formulaBool.toString() : "";
+      //     case "date":
+      //       return `${formulaProp[formulaProp.type]?.start ?? ""}${
+      //         formulaProp[formulaProp.type]?.end
+      //           ? `- ${formulaProp[formulaProp.type]?.end}`
+      //           : ""
+      //       }`;
+      //     case "number": {
+      //       const formulaNumber = formulaProp[formulaProp.type];
+      //       return formulaNumber !== null ? formulaNumber.toString() : "";
+      //     }
+      //     case "string":
+      //       return formulaProp[formulaProp.type] || "";
+      //   }
+      // }
+      case "title":
+        return prop[prop.type]
+          .map((v) =>
+            this.n2mClient.annotatePlainText(v.plain_text, v.annotations)
+          )
+          .join("");
+      case "rich_text":
+        return prop[prop.type]
+          .map((v) =>
+            this.n2mClient.annotatePlainText(v.plain_text, v.annotations)
+          )
+          .join("");
+      case "people":
+        return `[${prop[prop.type]
+          .map((v) => `["${v.object}", "${v.id}"]`)
+          .join(", ")}]`;
+      case "unique_id":
+        return `${prop[prop.type].prefix || ""}${prop[prop.type].number}`;
+      case "relation":
+        return `[${prop[prop.type].map((v) => `"${v.id}"`).join(", ")}]`;
+      // TODO: BLOCKER: Not all rollup types are supported in the SDK
+      // case "rollup": {
```

Comment:
Can we remove this comment or will it be a fast-follow?

Reply:
It's dependent on Notion-SDK. I'm assuming that it won't be quick. 

One option is to enable the sub types defined here (which is what the notion SDK supports) for rollups and functions. Then return unsupported type for any subtypes that aren't included here.

I decided against it because I think it would be confusing for a user to return some rollup/function sub types and not others. 
Just as it is confusing to me why this is the case in the Notion SDK. 😅

Reply:
Short answer, yes it can be removed.